SpatialGDK/Source/SpatialGDKEditor/Private/SpatialGDKDefaultLaunchConfigGenerator.cpp

```diff
@@ -101,9 +102,78 @@ bool WriteLoadbalancingSection(TSharedRef<TJsonWriter<>> Writer, const FName& Wo
 	return true;
 }
 
+} // anonymous namespace
+
+void SetLevelEditorPlaySettingsWorkerTypes(const TMap<FName, FWorkerLaunchConfiguration>& InWorkers)
+{
+	ULevelEditorPlaySettings* PlayInSettings = GetMutableDefault<ULevelEditorPlaySettings>();
+
+	PlayInSettings->WorkerTypesToLaunch.Empty(InWorkers.Num());
+	for (const auto& Worker : InWorkers)
+	{
+		PlayInSettings->WorkerTypesToLaunch.Add(Worker.Key, Worker.Value.WorkerLoadBalancing->GetNumberOfWorkersForPIE());
+	}
 }
 
-bool GenerateDefaultLaunchConfig(const FString& LaunchConfigPath, const FSpatialLaunchConfigDescription* InLaunchConfigDescription)
+bool GetLoadBalancingStrategyFromWorldSettings(UWorld& World, USpatialOSAbstractLoadBalancingStrategy*& OutStrategy, FIntPoint& OutWorldDimension)
+{
+	const ASpatialWorldSettings* WorldSettings = Cast<ASpatialWorldSettings>(World.GetWorldSettings());
+
+	if (!WorldSettings)
+	{
+		UE_LOG(LogSpatialGDKDefaultLaunchConfigGenerator, Error, TEXT("Missing SpatialWorldSettings on map %s"), *World.GetMapName());
```

Comment:
The SpatialWorldSetting part was done before I started to work on this. I'll take a look at the other error messages that were added to improve these, since they will be the first to be seen.

Reply:
Which World Settings file that's used is set in the Project Settings. We created 
 a SpatialWorldSettings which has a customer load-balancing strategy and locking policy because it allows users to specify different load-balancing strategies and locking policies per map.